cmd/knit/cmd/irqaff.go

```diff
@@ -0,0 +1,105 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2020 Red Hat, Inc.
+ */
+
+package cmd
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/spf13/cobra"
+
+	"github.com/openshift-kni/debug-tools/pkg/irqs"
+	softirqs "github.com/openshift-kni/debug-tools/pkg/irqs/soft"
+	"k8s.io/kubernetes/pkg/kubelet/cm/cpuset"
+)
+
+type irqAffOptions struct {
+	checkEffective bool
+	checkSoftirqs  bool
+}
+
+func newIRQAffinityCommand() *cobra.Command {
+	opts := &irqAffOptions{}
+	irqAff := &cobra.Command{
+		Use:   "irqaff",
+		Short: "show IRQ/softirq thread affinities",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return showIRQAffinity(cmd, opts, args)
+		},
+		Args: cobra.NoArgs,
+	}
+	irqAff.Flags().BoolVarP(&opts.checkEffective, "effective-affinity", "E", false, "check effective affinity.")
+	irqAff.Flags().BoolVarP(&opts.checkSoftirqs, "softirqs", "s", false, "check softirqs counters.")
+	return irqAff
+}
+
+func showIRQAffinity(cmd *cobra.Command, opts *irqAffOptions, args []string) error {
+	if opts.checkSoftirqs {
+		src, err := os.Open(filepath.Join(knitOpts.procFSRoot, "softirqs"))
+		if err != nil {
+			return fmt.Errorf("error reading softirqs from %q: %v", knitOpts.procFSRoot, err)
+		}
+		info, err := softirqs.ReadInfo(src)
+		src.Close()
+
+		if err != nil {
+			return fmt.Errorf("error parsing softirqs from %q: %v", knitOpts.procFSRoot, err)
+		}
+
+		dumpSoftirqInfo(info, knitOpts.cpus)
+	} else {
+		flags := uint(0)
+		if opts.checkEffective {
+			flags |= irqs.EffectiveAffinity
+		}
+
+		irqInfos, err := irqs.ReadInfo(knitOpts.procFSRoot, flags)
```

Comment:
Same here. And if these files have any documentation, would be welcome

Reply:
not much more than man 5 procfs - added as comment.
The content of the file is pretty human-friendly though.

Reply:
ok